File: lib/layer2_core/light/e_surface_dim_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ESurfaceDimLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl10;
      case 'standard': return cstl10;
      case 'blue': return cbl10;
      case 'green': return cgl10;
      case 'red': return crl10;
      case 'nothing': return cnol10;
      default: return cstl10;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ESurfaceDimLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceDimLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl10
            AppTheme.STANDARD -> R.color.cstl10
            AppTheme.BLUE -> R.color.cbl10
            AppTheme.GREEN -> R.color.cgl10
            AppTheme.RED -> R.color.crl10
            AppTheme.NOTHING -> R.color.cnol10
        }
    }
}
